apm-agent-plugins/apm-okhttp-plugin/src/main/java/co/elastic/apm/agent/okhttp/OkHttpClientAsyncInstrumentation.java

```diff
@@ -61,27 +61,32 @@ public String getAdviceClassName() {
         public static Object[] onBeforeEnqueue(final @Advice.Origin Class<? extends Call> clazz,
                                                final @Advice.FieldValue("originalRequest") @Nullable Request originalRequest,
                                                final @Advice.Argument(0) @Nullable Callback originalCallback) {
-            if (tracer.getActive() == null) {
+
+            final AbstractSpan<?> parent = tracer.getActive();
+            if (parent == null) {
                 return null;
             }
 
             if (originalRequest == null || originalCallback == null) {
                 return null;
             }
 
-            final AbstractSpan<?> parent = tracer.getActive();
-
             Request request = originalRequest;
             Callback callback = originalCallback;
             URL url = request.url();
+
             Span span = HttpClientHelper.startHttpClientSpan(parent, request.method(), url.toString(), url.getProtocol(),
                 OkHttpClientHelper.computeHostName(url.getHost()), url.getPort());
+
+            Request.Builder builder = originalRequest.newBuilder();
             if (span != null) {
                 span.activate();
-                Request.Builder builder = originalRequest.newBuilder();
                 span.propagateTraceContext(builder, OkHttpRequestHeaderSetter.INSTANCE);
                 request = builder.build();
                 callback = CallbackWrapperCreator.INSTANCE.wrap(originalCallback, span);
+            } else {
```

Comment:
[minor] maybe add a comment here (and in similar code) to explain that here `span == null` means that there is an exit span already active, and thus that we still need to propagate context of the parent.

Reply:
It's basically true now, but conceptually it can be something else. If, for whatever reason we decide no to create spans here, as long as we want to support distributed tracing, we need to propagate _some_ trace context headers, and in order for distributed tracing to work, it needs to be the parent's trace context, which we set as non-discardable.
If the parent is not an exit span, it will not be perfect and things like service map will not be populated properly, so we will have to see what we do with that if we find such, but for the purpose of context propagation - I believe this is what we should do.